### DISCUSSION TITLE
ws realtime results

### DISCUSSION THREAD

**Discussion by ghost** [searx/searx#3211] [Ideas]
I have an idea that will make searx feel snappier.
This may be a good idea for a plugin or core feature.

Right now the order of a search is something like
0. GET/POST search request 
1. gather from engines
2. push to client

this just feels rather slow because you're gathering results from several places

what I came up with was an order along the lines of
0. GET/POST search request
0.1. push template page with ws connection
1. a queue is followed and per engine a ws message is sent
1.1 when the ws receives a result (set) it adds it to the page and changes the client's result order to relevancy
2. after it went by every engine ws closes.

May be smart, may be stupid. 
I'd love to hear your opinions!

**Comment by unixfox** [comment]
This would benefit only people that are using javascript, a lot of people in the privacy community have javascript disabled.

I would also say that this would require a huge rewrite of the core of searx for that to happen.

**Reply by ghost** [reply]
ah, didn't think about that one haha!
well, t'was but an suggestion but too bad!
